python/src/init_hpx.cpp

```diff
@@ -16,17 +16,59 @@
 
 #if defined(linux) || defined(__linux) || defined(__linux__)
 
+#include <sstream>
+#include <fstream>
+#include <streambuf>
+
 int __argc = 0;
 char** __argv = nullptr;
 
-void set_argv_argv(int argc, char* argv[], char* env[])
+void set_argc_argv(int argc, char* argv[], char* env[])
 {
     __argc = argc;
     __argv = argv;
 }
 
 __attribute__((section(".init_array")))
-    void (*set_global_argc_argv)(int, char*[], char*[]) = &set_argv_argv;
+    void (*set_global_argc_argv)(int, char*[], char*[]) = &set_argc_argv;
+
+struct get_command_line
+{
+    get_command_line()
+    {
+        // parse the command line proc file
+        std::ifstream cmdline("/proc/self/cmdline");
+        std::string tmpstr((std::istreambuf_iterator<char>(cmdline)),
+                         std::istreambuf_iterator<char>());
+        // make a stringstream and split into a vector of strings
+        std::stringstream ss(tmpstr);
+        std::istream_iterator<std::string> begin(ss);
+        std::istream_iterator<std::string> end;
+        std::vector<std::string> args_(begin, end);
```

Comment:
Wouldn't it be possible to directly initialize the vector from the ifstream:

    std::ifstream cmdline("/proc/self/cmdline"); 
    if (!cmdline.is_open())
    {
        // proc fs is not available, raise error etc.
    }
    std::vector<std::string> args{
        std::istream_iterator<char>{cmdline},  
        std::istream_iterator<char>{}; 
    };

?

Reply:
Yes, that is easier. :)

But I had to change the {} to () to get past the "most vexing parse" problem...